src/sentry_plugins/clubhouse/plugin.py

```diff
@@ -0,0 +1,95 @@
+
+    description = 'Create Clubhouse Stories from a project.'
+    slug = 'clubhouse'
+    title = 'Clubhouse'
+    conf_title = title
+    conf_key = 'clubhouse'
+
+    def get_group_urls(self):
+        return super(ClubhousePlugin, self).get_group_urls() + [
+            (
+                r'^autocomplete', IssueGroupActionEndpoint.as_view(
+                    view_method_name='view_autocomplete',
+                    plugin=self,
+                )
+            ),
+        ]
+
+    def get_configure_plugin_fields(self, request, project, **kwargs):
+        token = self.get_option('token', project)
+        helptext = (
+            'Enter your API Token (found on '
+            'your account Settings, under API Tokens).'
+        )
+        secret_field = get_secret_field_config(token, helptext, include_prefix=True)
+        secret_field.update(
+            {
+                'name': 'token',
+                'label': 'API Token',
+                'placeholder': 'e.g. 12345678-1234-1234-1234-1234567890AB'
+            }
+        )
+        return [
+            secret_field, {
+                'name': 'project',
+                'label': 'Project ID',
+                'default': self.get_option('project', project),
+                'type': 'text',
+                'placeholder': 'e.g. 639281',
+                'help': 'Enter your project\'s numerical ID.'
+            }
+        ]
+
+    def is_configured(self, request, project, **kwargs):
+        return all(self.get_option(k, project) for k in ('token', 'project'))
+
+    def create_issue(self, request, group, form_data, **kwargs):
+        client = ClubhouseClient()
+
+        try:
+            response = client.create_story(
+                project=self.get_option('project', group.project),
+                data=form_data,
+                token=self.get_option('token', group.project)
+            )
+        except Exception as e:
+            self.raise_error(e)
+
+        return response['id']
+
+    def get_issue_label(self, group, issue_id, **kwargs):
+        return 'Clubhouse Story #%s' % issue_id
+
+    def get_issue_url(self, group, issue_id, **kwargs):
+        # TODO: Uncertain of this is a good method, as it incurs an API call on page load.
+        # `issue_url` can't be generated purely from a knonwn string + `issue_id`, as it
+        # requires the Clubhouse Organization in the URL, which isn't currently a known API call.
+        # Can the `app_url` be persisted/called somewhere, when the link is created?
+        client = ClubhouseClient()
+        story = client.get_story(issue_id, token=self.get_option('token', group.project))
+        return story['app_url']
+
+    def validate_config(self, project, config, actor):
+        try:
+            config['project'] = int(config['project'])
+        except ValueError as exc:
+            self.logger.exception(six.text_type(exc))
+            raise PluginError('Invalid project value, numbers only')
```

Comment:
Is this descriptive enough? The msg confuses me. :)

Reply:
Probably not.  It was the last thing I did, and I don't have a good test case for it either.
I think a more evolved plugin could perform a `list` of projects, allow the user to type in a name, and then retrieve and store the `id`, preventing the user from seeing numbers at all.
But for now, a better error message might be:

> Invalid Project ID. Project IDs are numbers-only, and can be found on the Project's page

Trying to keep it concise.

Reply:
+1 to that message. 

Yeah I started to add "this needs a test" but... :(